Sources/App/Models/Package.swift

```diff
@@ -45,6 +51,27 @@ final class Package: Model, Content {
 }
 
 
+extension Package {
+    var repository: Repository? {
+        repositories.first
+    }
+}
+
+
+extension Package {
+    var defaultVersion: Version? {
+        // TODO: sas 2020-04-30: find a more convenient way to use this. In order to avoid
+        // fatalErrors from lack of lazy loading, the caller needs to use it on a Package that's
+        // been fetched like so:
+        //   Package.query(on: db).with(\.$versions).with(\.$repositories)
+        // That's awkward. Should instead defaultBranch take a parameter (on: db) and do this
+        // itself?
+        guard let defaultBranch = repository?.defaultBranch else { return nil }
```

Comment:
A comment on default branches: Fetching default branch from the GitHub API isn't strictly necessary if we're pulling the entire source locally. The default branch will be selected after a fresh pull, and we shouldn't be switching branches in our local copy of the repository. Or, if you do want to verify it then it can be read out of the local git repository rather than relying on the GitHub API for this.

Reply:
Is there any reason to suspect the GH api to be wrong? Because we've already got the default branch info so it's easy to use it.

There's also the remote edge case that we pull a repo and the default branch changes afterwards. We'd sit on a wrong branch then unless we always ensure we track the default one as well, right?

Reply:
But see below, I'd probably rather ditch/ignore this for now!

Reply:
> Is there any reason to suspect the GH api to be wrong? Because we've already got the default branch info so it's easy to use it.

Only that it's more work when we support GitLab, which we already have a case for as there are some packages already not being indexed by that lack of support. If we use the local repository, it works for all services.

Of course, it's likely that the GitLab API also supplies the default branch, so it's not a huge issue. I'm happy either way 👍

> There's also the remote edge case that we pull a repo and the default branch changes afterwards. We'd sit on a wrong branch then unless we always ensure we track the default one as well, right?

Oof, that's an excellent point, and you just made me realise my current implementation is vulnerable to this! I just switch to the default branch on clone and only ever pull changes after that. Regardless of which method we use to determine the default branch, we should make sure to check it out before pulling when updating.

Reply:
WRT to Gitlab - the `projects` api resource seems to carry everything we're currently pulling in from Github: https://docs.gitlab.com/ee/api/projects.html

Reply:
Then let's just lean on the default branch from those API calls then. It's going to be a definitive source of truth, and I can't really see people needing more services than GitHub/Lab. Maybe BitBucket, but it feels like that's well and truly lost the open-source battle to those other two.

Reply:
Yeah, that one's gone =) 👍